src/app/_components/Event/EventSection.js

```diff
@@ -0,0 +1,154 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import { useInView } from 'react-intersection-observer';
+import { useSearchParams } from 'next/navigation';
+import { CircularProgress } from '@mui/material';
+import CheckMark from '@icons/check-mark.svg';
+import Search from '@icons/search.svg';
+import { allEvents, useSetParam } from '@hooks';
+import { EventCard } from '@components/Event/Card';
+import { PillButton } from '@components/PillButton';
+import { buttonColorVariant } from '@components/PillButton/style';
+import { NoInfoToShow } from '@components/NoInfoToShow';
+import { cn } from '@/utils/cn';
+
+const currentMonth = new Date().getMonth() + 1;
+
+const monthNames = [
+  { index: 0, name: 'січень', en: 'january' },
+  { index: 1, name: 'лютий', en: 'february' },
+  { index: 2, name: 'березень', en: 'march' },
+  { index: 3, name: 'квітень', en: 'april' },
+  { index: 4, name: 'травень', en: 'may' },
+  { index: 5, name: 'червень', en: 'june' },
+  { index: 6, name: 'липень', en: 'july' },
+  { index: 7, name: 'серпень', en: 'august' },
+  { index: 8, name: 'вересень', en: 'september' },
+  { index: 9, name: 'жовтень', en: 'october' },
+  { index: 10, name: 'листопад', en: 'november' },
+  { index: 11, name: 'грудень', en: 'december' },
+];
+
+const calcPureMonthIndex = parseInt(currentMonth - 1, 10);
+const firstMonth = monthNames.slice(calcPureMonthIndex, calcPureMonthIndex + 1);
+const restMonth = monthNames.slice(calcPureMonthIndex + 1, calcPureMonthIndex + 6);
+const filteredMonths = firstMonth.concat(restMonth);
+
+export function EventSection() {
+  const [activeMonth, setActiveMonth] = useState(parseInt(filteredMonths[0].index + 1, 10));
+
+  const searchParams = useSearchParams();
+  const { add: addParam, remove: removeParam } = useSetParam('month');
+
+  const { ref, inView } = useInView();
+
+  const { semiorange } = buttonColorVariant.eventFilter;
+
+  const activeButtonStyles = cn({
+    'pointer-events-none border-secondary-300 bg-secondary-300 font-semibold text-gray-900': semiorange.active,
+  });
+
+  function selectButtonIcon(index) {
+    if (activeMonth - 1 === index) {
+      return (
+        <CheckMark
+          key={`checkmark+${index}`}
+          className={cn(`block h-4 w-4 transition-all group-hover:hidden group-focus:hidden`)}
+        />
+      );
+    }
+
+    return (
+      <Search
+        key={`searchicon+${index}`}
+        className="hidden h-4 w-4 transition-all group-hover:block group-focus:block"
+      />
+    );
+  }
```

Comment:
@Yivgha I would inline this logic into PillButton, otherwise you should wrap it with useCallback

Reply:
changed